### PR TITLE
Don't locate DbContexts and DbMigrationsConfigurations that cannot be constructed

### PR DESCRIPTION
**Pull request by nphmuller**
When an abstract (or generic) base class is made for either types, they might be used, instead of the implementation.

Example: I have an abstract class DbContextBase and a class MyApplicationContext : DbContextBase.
Migrator.EF6 would incorrectly detect DbContextBase as the single true DbContext, instead of MyApplicationContext.

The constraint I've used are taken from the [EF source code](https://entityframework.codeplex.com/SourceControl/latest#src/EntityFramework/Migrations/Utilities/MigrationsConfigurationFinder.cs)

### CODE REVIEW

File: src/Migrator.EF6.Tools/Executor.cs [hunk 1]
```diff
@@ -67,7 +67,11 @@ public void EnableMigrations(string outputDir)
 
 		private string FindAppDbContextTypeName()
 		{
-			var allDbContextTypes = _types.Where(t => typeof(DbContext).IsAssignableFrom(t));
+			var allDbContextTypes = _types.Where(
+				t => typeof(DbContext).IsAssignableFrom(t)
```
Comment (mrahhal) [inline src/Migrator.EF6.Tools/Executor.cs]: Can we make this a method and use it in `FindDbMigrationsConfiguration` too? (it's duplicated)
Perhaps `var allDbContextTypes = GetConstructablesOfType(typeof(DbContext));`.

**Comment by mrahhal** [comment]
And @nphmuller, for your info, EF6 is now maintained at GitHub instead of codeplex! Here: [aspnet/EntityFramework6](https://github.com/aspnet/EntityFramework6).

File: src/Migrator.EF6.Tools/Executor.cs (hunk 1 above)
Comment (nphmuller) [reply src/Migrator.EF6.Tools/Executor.cs]: Yeah, I think they did that in the EF source code too. Will add it in a minute.

**Comment by nphmuller** [comment]
Thanks for the link!
I moved the duplicate type checking to an extension method. Let me know what you think!

Edit: Ah, I misread your suggestion. If you think a method like GetConstructablesOfType() is better, I'll rewrite it.

**Comment by mrahhal** [comment]
Looks great. But I think you used spaces in `Executor.cs` and in `TypeExtensions.cs` (I'm using tabs). Please turn them to tabs and I'll merge (`ctrl+r ctrl+w` will let you see whitespace).
Also, I recommend using this extension for visual studio so that you don't have to switch between spaces/tabs if the project you're contributing to has different set of conventions: [editorconfig](http://editorconfig.org).

**Comment by nphmuller** [comment]
Should be all tabs now ;)
I'll take a look at the extension. Sounds really handy.

**Comment by mrahhal** [comment]
Thanks a lot!

**Pull request merged**